apps/wallet-api-tools/src/Editor.tsx

```diff
@@ -1,92 +1,180 @@
 "use client";
-
+import { useAtom } from "jotai";
+import { atomWithStorage } from "jotai/utils";
 import { Transport, WindowMessageTransport } from "@ledgerhq/wallet-api-core";
 import {
   getSimulatorTransport,
   profiles,
 } from "@ledgerhq/wallet-api-simulator";
-import { langs } from '@uiw/codemirror-extensions-langs';
-import CodeMirror, { ReactCodeMirrorRef } from "@uiw/react-codemirror";
 import { useSearchParams } from "next/navigation";
-import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { Fragment, useCallback, useEffect, useRef, useState } from "react";
 import { v4 as uuidv4 } from "uuid";
-import { z } from "zod";
 import { Input } from "./Input";
+import {
+  MessageGrouped,
+  Response,
+  MessageOut,
+  MessageInfo,
+  ResponseResult,
+  ResponseError,
+  schemaRequest,
+  Request,
+  MessageIn,
+  schemaResponse,
+} from "./types";
+import GroupedMessage from "./components/GroupedMessage";
+import InfoMessage from "./components/InfoMessage";
+import { ArrowDown } from "lucide-react";
 
-const schemaRequest = z.object({
-  method: z.string(),
-  params: z.object({}).passthrough(),
-});
-
-type Message = {
-  type: "in" | "out" | "info" | "error";
-  value: string;
-  date: Date;
-};
-
-function getMessageStatus(message: Message) {
-  switch (message.type) {
-    case "in":
-      return "<- [received]";
-    case "out":
-      return "-> [sent]";
-    case "error":
-      return "[error]";
-    case "info":
-      return "[info]";
-    default: {
-      const exhaustiveCheck: never = message.type; // https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking
-      return exhaustiveCheck;
-    }
-  }
-}
+const historyAtom = atomWithStorage<(MessageGrouped | MessageInfo)[]>(
+  "history",
+  [],
+);
 
 export function Editor() {
-  const [history, setHistory] = useState<Message[]>([]);
+  const [history, setHistory] = useAtom(historyAtom);
   const searchParams = useSearchParams();
-  const codeMirrorRef = useRef<ReactCodeMirrorRef | null>(null);
+  const panelRef = useRef<HTMLDivElement>(null);
   const transportRef = useRef<Transport | null>(null);
+  const [value, setValue] = useState<string>("");
+  const [newElement, setNewElement] = useState<boolean>(false);
+  const [scrollBottom, setScrollBottom] = useState(true);
+
+  const handleReUse = (request: MessageOut) => {
+    const parsedSentValue: Request = request.value;
 
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
We can remove this comment by editing `apps/wallet-api-tools/.eslintrc.js` and adding this to the rules object:
```js
    "@typescript-eslint/no-unused-vars": [
      "error",
      { ignoreRestSiblings: true },
    ],
```